std/assembly/internal/sort.ts

```diff
@@ -36,8 +36,12 @@ export function COMPARATOR<T>(): (a: T, b: T) => i32 {
     }
   } else if (isString<T>()) {
     return (a: T, b: T): i32 => {
-      var sa = <string>a, sb = <string>b;
-      return compareUnsafe(sa, 0, sb, 0, min(sa.length, sb.length));
+      if (a === b || a === null || b === null) return 0;
+      var alen = (<string>a).length;
+      var blen = (<string>b).length;
+      if (!alen) return -1;
```

Comment:
Not sure about `if (alen == alen && alen == 0) return 0`. This works well without this check, but potentially make sorting unstable.

Reply:
Not sure about the additional case, seems that we don't need it. Or does it prevent issues?

Reply:
it potentially prevent unstable sorting when we compare one empty string `""` with another empty string and they are could potentially swap and change order. Currently all browser engines switches to stable sort based on TimSort and I also have plan rewrite existing sort algorithm to TimSort or GrailSort (both stable)